From 84648eb290c22f61dc58a9626c16b30f2177272f Mon Sep 17 00:00:00 2001
From: An Guoli <qaq@qaq.land>
Date: Sun, 20 Sep 2026 02:37:41 +0800
Subject: [PATCH] add linux/stddef.h for musl compatibility (#175)

The __always_inline macro is defined in <linux/stddef.h>. While glibc
includes this by default, musl does not. This commit explicitly includes
<linux/stddef.h> to ensure compatibility with musl libc platforms.
---
 include/scenefx/types/fx/clipped_region.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/scenefx/types/fx/clipped_region.h b/include/scenefx/types/fx/clipped_region.h
index 02bac5dc0..714459128 100644
--- a/include/scenefx/types/fx/clipped_region.h
+++ b/include/scenefx/types/fx/clipped_region.h
@@ -1,6 +1,7 @@
 #ifndef SCENEFX_TYPES_FX_CLIPPED_REGION_H
 #define SCENEFX_TYPES_FX_CLIPPED_REGION_H
 
+#include <linux/stddef.h>
 #include <wlr/util/box.h>
 
 // uint16_t is a reasonable enough range for corner radius
